### PR TITLE
Doxygen section support

### LINKED ISSUE #938: Mirror Doxygen sections in Haskell export list
If we generate an export list (#76), and if the C header uses Doxygen sections, it would be nice to use those same sections also in the export list.

### PR DESCRIPTION
**Pull request by bolt12**
Closes #938

### CODE REVIEW

File: hs-bindgen/examples/golden/documentation/doxygen_docs.h
Comment (bolt12) [inline hs-bindgen/examples/golden/documentation/doxygen_docs.h:1]: Make this example more exhaustive, follow: https://www.doxygen.nl/manual/grouping.html

**Comment by bolt12** [comment]
I need to look for real world usages of these doxygen commands, and go from there. Test if we can parse them, check for more edge cases.

File: hs-bindgen/examples/golden/documentation/doxygen_docs.h
Comment (bolt12) [inline hs-bindgen/examples/golden/documentation/doxygen_docs.h:1]: forward_declared_struct and union should be renamed to opaque

File: hs-bindgen/src-internal/HsBindgen/Backend/HsModule/Translation.hs [hunk 1]
```diff
@@ -56,6 +63,25 @@ data ImportListItem =
...
+-- > [ ExportEntry (ExportName "api_version")     -- ungrouped, hoisted
+-- > , ExportSectionHeader 1 "Core Data Types"    -- section transition
+-- > , ExportEntry (ExportTypeAll "Config_t")      -- grouped
+-- > , ExportEntry (ExportPattern "COLOR_RED")     -- derived from grouped
+-- > ]
+data ExportEntry =
+    ExportEntry ExportItem
+    -- | A Haddock section header at the given depth.  Depth 1 means @-- *@,
+    -- depth 2 means @-- **@, etc.
+  | ExportSectionHeader Natural Text
```
Comment (bolt12) [inline hs-bindgen/src-internal/HsBindgen/Backend/HsModule/Translation.hs]:
```suggestion
data ExportEntry = 
    ExportEntry ExportItem
  | ExportSection Text [ExportEntry]
```

File: hs-bindgen/src-internal/HsBindgen/Backend/HsModule/Translation.hs
Comment (bolt12) [inline hs-bindgen/src-internal/HsBindgen/Backend/HsModule/Translation.hs:1]: Move some of this logic to a Doxygen specific module and hide internal types there.

File: hs-bindgen/src-internal/HsBindgen/Backend/HsModule/Translation.hs [hunk 2]
```diff
@@ -391,14 +453,208 @@ resolveExtHsRefImports extRef = ImportAcc{
...
+--     from anonymous\/nested\/derived ones ('ExportDerived').
+--
+-- Example: given @\@defgroup core "Core"@ containing @config_t@ and its
+-- anonymous inner union, plus an ungrouped @api_version_t@:
+--
+-- > Config_t        → ExportGrouped ["Core"]    (step 1: Haskell name hit)
+-- > Config_t_union  → ExportGrouped ["Core"]    (step 2: C name "config_t" hit)
+-- > Api_version_t   → ExportUngrouped           (step 3: top-level, no group)
+-- > pattern COLOR_RED → ExportDerived            (step 3: not top-level)
+data ExportGroupTag =
```
Comment (bolt12) [inline hs-bindgen/src-internal/HsBindgen/Backend/HsModule/Translation.hs]: This is an internal type

File: lychee.toml
Comment (bolt12) [inline lychee.toml:1]: Remove this

**Review by edsko — APPROVED** [review]
Nice! Feel free to merge one that data type is changed and we have a test for section nesting.
